cmd/thermal-recorder/frameloop.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 The Cacophony Project. All rights reserved.
+// Use of this source code is governed by the Apache License Version 2.0;
+// see the LICENSE file for further details.
+
+package main
+
+import (
+	"github.com/TheCacophonyProject/lepton3"
+)
+
+type LeptonFrameWriter interface {
+	WriteFrame(prevFrame, frame *lepton3.Frame) error
+}
+
+func NewFrameLoop(size int) *FrameLoop {
+	frames := make([]*lepton3.Frame, size)
+	for i := range frames {
+		frames[i] = new(lepton3.Frame)
+	}
+
+	return &FrameLoop{
+		size:         size,
+		currentIndex: 0,
```

Comment:
In Go, all values have a "zero value" if not otherwise initialised, so this line isn't strictly necessary.